tests/kernel/threads/thread_error_case/src/test_thread_error.c

```diff
@@ -0,0 +1,236 @@
+{
+	int choice = *((int *)p1);
+	uint32_t perm = K_INHERIT_PERMS;
+
+	TC_PRINT("current case is %d\n", choice);
+
+	/* Set up the fault or assert are expected before we call
+	 * the target tested funciton.
+	 */
+	switch (choice) {
+	case THREAD_START:
+		ztest_set_fault_valid(true);
+		k_thread_start(NULL);
+		break;
+	case FLOAT_DISABLE:
+		ztest_set_fault_valid(true);
+		k_float_disable(NULL);
+		printk("!!!!!!!!!!!!!!!\n");
+		break;
+	case TIMEOUT_REMAINING_TICKS:
+		ztest_set_fault_valid(true);
+		k_thread_timeout_remaining_ticks(NULL);
+		break;
+	case TIMEOUT_EXPIRES_TICKS:
+		ztest_set_fault_valid(true);
+		k_thread_timeout_expires_ticks(NULL);
+		break;
+	case THREAD_CREATE_NEWTHREAD_NULL:
+		ztest_set_fault_valid(true);
+		if (_is_user_context()) {
+			perm = perm | K_USER;
+		}
+
+		k_thread_create((struct k_thread *)NULL, test_stack, STACK_SIZE,
+			test_thread, NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			perm, K_NO_WAIT);
+		break;
+	case THREAD_CREATE_STACK_NULL:
+		ztest_set_fault_valid(true);
+		if (_is_user_context()) {
+			perm = perm | K_USER;
+		}
+
+		k_thread_create(&test_tdata, NULL, STACK_SIZE,
+			test_thread, NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			perm, K_NO_WAIT);
+		break;
+	case THREAD_CTEATE_STACK_SIZE_OVERFLOW:
+		ztest_set_fault_valid(true);
+		if (_is_user_context()) {
+			perm = perm | K_USER;
+		}
+		k_thread_create(&test_tdata, test_stack, -1,
+			test_thread, NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			perm, K_NO_WAIT);
+		break;
+	default:
+		TC_PRINT("should not be here!\n");
+		break;
+	}
+
+	/* If negative comes here, it means error condition not been
+	 * detected.
+	 */
+	ztest_test_fail();
+}
+
+static void create_negative_test_thread(int choice)
+{
+	int ret;
+	uint32_t perm = K_INHERIT_PERMS;
+
+	if (_is_user_context()) {
+		perm = perm | K_USER;
+	}
+
+	case_type = choice;
+
+	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
+			(k_thread_entry_t)tThread_entry_negative,
+			(void *)&case_type, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			perm, K_NO_WAIT);
+
+	ret = k_thread_join(tid, K_FOREVER);
+}
+#endif
+
+#ifdef CONFIG_USERSPACE
+/* TESTPOINT: Pass a null pointer into the API k_thread_start() */
+static void test_thread_start(void)
+{
+	create_negative_test_thread(THREAD_START);
+}
+#else
+static void test_thread_start(void)
+{
+	ztest_test_skip();
+}
+#endif
+
+#ifdef CONFIG_USERSPACE
+/* TESTPOINT: Pass a null pointer into the API k_float_disable() */
+static void test_float_disable(void)
+{
+	create_negative_test_thread(FLOAT_DISABLE);
+}
+#else
+static void test_float_disable(void)
+{
+	ztest_test_skip();
+}
+#endif
+
+#ifdef CONFIG_USERSPACE
+/* TESTPOINT: Pass a null pointer into the API */
+static void test_timeout_remaining_ticks(void)
+{
+	create_negative_test_thread(TIMEOUT_REMAINING_TICKS);
+}
+#else
+static void test_timeout_remaining_ticks(void)
+{
+	ztest_test_skip();
```

Comment:
I think write this way for dedicated few test cases is ok. But it seems like all your test cases in this files are under userspace. If so, my suggestion is remove all this #if-else and modify the testcase.yaml instead.

Reply:
Yes, I have added filter and remove the duplicate code.